### PR TITLE
feat: web3-profile-plugin

### PR DESCRIPTION
**Pull request by Randolph314**
## Description

Closes [#282](https://mask.atlassian.net/browse/MF-282)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -101,6 +101,8 @@
       "@masknet/plugin-cyberconnect": ["./packages/plugins/CyberConnect/src"],
       "@masknet/plugin-go-plus-security": ["./packages/plugins/GoPlusSecurity/src"],
       "@masknet/plugin-cross-chain-bridge": ["./packages/plugins/CrossChainBridge/src"],
+      "@masknet/plugin-web3-profile": ["./packages/plugins/Web3Profile/src"],
+      "@masknet/plugin-web3-profile": ["./packages/plugins/Web3Profile/src"],
```
Comment (Jack-Works) [inline tsconfig.json:105]: Duplicated.

File: packages/plugins/tsconfig.json [hunk 2]
```diff
@@ -10,6 +10,8 @@
     { "path": "./RSS3/" },
     { "path": "./DAO/" },
     { "path": "./CyberConnect/" },
-    { "path": "./CrossChainBridge/" }
+    { "path": "./CrossChainBridge/" },
+    { "path": "./Web3Profile/" },
+    { "path": "./Web3Profile/" }
```
Comment (Jack-Works) [inline packages/plugins/tsconfig.json:15]: Duplicated.

File: packages/mask/src/social-network/ui.ts [hunk 3]
```diff
@@ -130,7 +138,9 @@ export async function activateSocialNetworkUIInner(ui_deferred: SocialNetworkUI.
                 },
                 personaSign: Services.Identity.signWithPersona,
                 walletSign: Services.Ethereum.personalSign,
+                generateSign: Services.Identity.generateSignResult,
```
Comment (Jack-Works) [inline packages/mask/src/social-network/ui.ts:148]: Rename those 3 methods? They're too similar on the name

File: packages/plugins/Web3Profile/src/SNSAdaptor/hooks/usePersonaSign.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { context } from '../context'
+
+export function usePersonaSign() {
+    return context.generateSign
```
Comment (Jack-Works) [inline packages/plugins/Web3Profile/src/SNSAdaptor/hooks/usePersonaSign.ts]: 🤔 can you use the context directly? This hook looks like does nothing.

File: .i18n-codegen.json [hunk 5]
```diff
@@ -218,14 +218,26 @@
...
+        },
         {
             "input": "./packages/mask/src/plugins/Tips/locales/en-US.json",
             "output": "./packages/mask/src/plugins/Tips/locales/i18n_generated",
             "parser": "i18next",
             "generator": {
                 "type": "i18next/react-hooks",
                 "hooks": "useI18N",
-                "namespace": "com.maskbook.tip",
+                "namespace": "io.mask.web3-profile-plugin",
```
Comment (Lanttcat) [inline .i18n-codegen.json:240]: Duplicated namespace  `o.mask.web3-profile-plugin`.

**Pull request closed without merging**